eng/pipelines/templates/jobs/sdk-diff-tests.yml

```diff
@@ -0,0 +1,136 @@
+      if [[ -z "$installer_build" ]]; then
+        fail "Could not find a build of installer for commit '$installer_sha'"
+        exit 1
+      fi
+
+      echo "Installer build: https://dev.azure.com/dnceng/internal/_build/results?buildId=$installer_build&view=results"
+
+      echo "##vso[build.addbuildtag]installer-$installer_sha"
+      echo "##vso[task.setvariable variable=InstallerBuildId]$installer_build"
+      echo "##vso[task.setvariable variable=DotnetDotnetBuildId]$dotnet_dotnet_build"
+    displayName: Find associated builds
+    name: Get_Build_Ids
+    env:
+      AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
+
+  - task: DownloadPipelineArtifact@2
+    displayName: Download MSFT SDK
+    inputs:
+      buildType: specific
+      buildVersionToDownload: specific
+      project: internal
+      pipeline: $(INSTALLER_OFFICIAL_CI_PIPELINE_ID)
+      buildId: $(InstallerBuildId)
+      artifact: BlobArtifacts
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-@(alpha|preview|rc|rtm)*)-linux-${{ parameters.architecture }}.tar.gz' 
+      allowPartiallySucceededBuilds: true
+      allowFailedBuilds: true
+      downloadPath: $(Pipeline.Workspace)/Artifacts
+      checkDownloadedFiles: true
+
+  - task: DownloadPipelineArtifact@2
+    displayName: Download Source Build SDK
+    inputs:
+      buildType: specific
+      buildVersionToDownload: specific
+      project: internal
+      pipeline: $(DOTNET_DOTNET_CI_PIPELINE_ID)
+      buildId: $(DotnetDotnetBuildId)
+      artifact: ${{ parameters.buildName }}_${{ parameters.architecture }}_Artifacts
+      patterns: '**/dotnet-sdk-+([0-9]).+([0-9]).+([0-9])?(-@(alpha|preview|rc|rtm)*)-${{ parameters.targetRid }}.tar.gz' 
+      allowPartiallySucceededBuilds: true
+      allowFailedBuilds: true
+      downloadPath: $(Pipeline.Workspace)/Artifacts
+      checkDownloadedFiles: true
+    
+  - script: |
+      set -x
+
+      msft_sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-linux-${{ parameters.architecture }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z $msft_sdk_tarball_name ]]; then
+        fail "Microsoft SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'"
+        exit 1
+      fi
+
+      sdk_tarball_name=$(find "$(Pipeline.Workspace)/Artifacts" -name "dotnet-sdk-*-${{ parameters.targetRid }}.tar.gz" -exec basename {} \;)
+
+      if [[ -z $sdk_tarball_name ]]; then
+        fail "Source-build SDK tarball does not exist in '$(Pipeline.Workspace)/Artifacts'"
+        exit 1
+      fi
+
+      eng/common/build.sh --projects $(Build.SourcesDirectory)/src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/Microsoft.DotNet.SourceBuild.SmokeTests.csproj --restore
+
+      echo "##vso[task.setvariable variable=MsftSdkTarballName]$msft_sdk_tarball_name"
+      echo "##vso[task.setvariable variable=SdkTarballName]$sdk_tarball_name"

```

Comment:
Since this task already works with the final location `$(Pipeline.Workspace)/Artifacts`, it would be better if you exported the whole path right away and just consume it in the next step